### ISSUE TITLE
withObjectType option don't support union type

### ISSUE THREAD

**Issue by anhdd-kuro** [Code-Hex/graphql-codegen-typescript-validation-schema#181]
I have gql file like this

```gql
type WorkDivision {
  id: ID!
  name: String!
}

type AttendanceNotEnteredOrDayOff {
  id: ID!
  name: String!
  symbol: String!
}

union AttendanceType = WorkDivision | AttendanceNotEnteredOrDayOff

type Attendance {
  date: String!
  type: AttendanceType!
}

```

**current output**

```ts
export function AttendanceSchema(): z.ZodObject<Properties<Attendance>> {
  return z.object({
    __typename: z.literal('Attendance').optional(),
    date: z.string().min(1),
    type: definedNonNullAnySchema // wrong !
  })
}
```

**expected output**

```ts
export function AttendanceTypeSchema(): z.ZodUnion<[z.ZodObject<Required<Properties<WorkDivision>>>, z.ZodObject<Required<Properties<AttendanceNotEnteredOrDayOff>>>]> {
  return z.union([WorkDivisionSchema(), AttendanceNotEnteredOrDayOffSchema()])
}

export function AttendanceSchema(): z.ZodObject<Properties<Attendance>> {
  return z.object({
    __typename: z.literal('Attendance').optional(),
    date: z.string().min(1),
    type: AttendanceTypeSchema()
  })
}
```

**Comment by Code-Hex** [comment]
@anhdd-kuro Thank you for your creating a new issue!
As you know, Basically this plugin is used for input schema mainly. So I don't have the motivation to add some types related to object schema. (like union, fragments, etc...)

It's OK to add some fixes on my side but It's a low priority. However, I appreciate it if you send the PR!

**Comment by Code-Hex** [comment]
@anhdd-kuro   published as v0.7.0
